back/app/policies/internal_comment_policy.rb

```diff
@@ -45,26 +45,22 @@ def permitted_attributes_for_create
   end
 
   def permitted_attributes_for_update
-    record.author_id == user&.id ? [:body_text] : []
+    internal_comment_author? ? [:body_text] : []
   end
 
   private
 
   def internal_commenter?
     if record.post_type == 'Idea'
-      active_admin_or_moderator?
+      active? && (admin? || UserRoleService.new.can_moderate?(record.post, user))
     elsif record.post_type == 'Initiative'
       active? && admin?
     else
       false
     end
   end
 
-  def active_admin_or_moderator?
-    active? && (admin? || UserRoleService.new.can_moderate?(record.post, user))
-  end
-
   def internal_comment_author?
-    record.author_id == user.id
+    record.author_id == user&.id
```

Comment:
I would prefer to explicitly return false if the `user` is `nil`. Otherwise, visitors could be mistakenly be considered authors of internal comments where the `author_id` is `nil`. I can imagine this happening for comments whose author has deleted their account.